pkg/client/clientset/typed/nodeset/v1alpha1/nodeclass.go

```diff
@@ -56,11 +56,42 @@ func newNodeClasses(c *NodesetV1alpha1Client) *nodeClasses {
 	}
 }
 
+// Get takes name of the nodeClass, and returns the corresponding nodeClass object, and an error if there is any.
+func (c *nodeClasses) Get(name string, options v1.GetOptions) (result *v1alpha1.NodeClass, err error) {
+	result = &v1alpha1.NodeClass{}
+	err = c.client.Get().
+		Resource("nodeclasses").
+		Name(name).
+		VersionedParams(&options, scheme.ParameterCodec).
+		Do().
+		Into(result)
+	return
+}
+
+// List takes label and field selectors, and returns the list of NodeClasses that match those selectors.
+func (c *nodeClasses) List(opts v1.ListOptions) (result *v1alpha1.NodeClassList, err error) {
+	result = &v1alpha1.NodeClassList{}
+	err = c.client.Get().
+		Resource("nodeclasses").
+		VersionedParams(&opts, scheme.ParameterCodec).
+		Do().
+		Into(result)
+	return
+}
+
+// Watch returns a watch.Interface that watches the requested nodeClasses.
+func (c *nodeClasses) Watch(opts v1.ListOptions) (watch.Interface, error) {
+	opts.Watch = true
+	return c.client.Get().
+		Resource("nodeclasses").
+		VersionedParams(&opts, scheme.ParameterCodec).
+		Watch()
+}
+
 // Create takes the representation of a nodeClass and creates it.  Returns the server's representation of the nodeClass, and an error, if there is any.
 func (c *nodeClasses) Create(nodeClass *v1alpha1.NodeClass) (result *v1alpha1.NodeClass, err error) {
 	result = &v1alpha1.NodeClass{}
 	err = c.client.Post().
-		Namespace(v1alpha1.TPRNamespace).
```

Comment:
TPRs will not exist anymore in 1.8, and are already deprecated in 1.7. We have to.